From cea2f84ede530545f787361ca97122446999db29 Mon Sep 17 00:00:00 2001
From: yzha108 <yanli.l.zhang@intel.com>
Date: Tue, 26 Jun 2018 19:55:38 +0800
Subject: [PATCH 2002/2345] media: intel-ipu4: max9286: AR0231 misses 2 lines

Message for Open Source:
Set max96705 VSYNC Regs

Message for Internal:
[Issue/Feature]
The last 2 lines pixel values are missing

[Root Cause/Changes]
Set max96705 VSYNC Regs

Change-Id: I95c1494dde286b1bcdf34d120e6ac131e878d08c
TrackedOn: #JIIAP-520
Signed-off-by: yzha108 <yanli.l.zhang@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/i2c/max9286-reg-settings.h |  6 ++++++
 drivers/media/i2c/max9286.c              | 25 +++++++++++++++++++++++++
 2 files changed, 31 insertions(+)

diff --git a/drivers/media/i2c/max9286-reg-settings.h b/drivers/media/i2c/max9286-reg-settings.h
index e1a2b0d..75dba98 100644
--- a/drivers/media/i2c/max9286-reg-settings.h
+++ b/drivers/media/i2c/max9286-reg-settings.h
@@ -44,6 +44,12 @@
 #define S_I2C_SOURCE_SER          0x0B
 #define S_I2C_DST_SER             0x0C
 #define S_INPUT_STATUS            0x15
+#define S_SYNC_GEN_CONFIG         0x43
+#define S_VS_DLY_2                0x44
+#define S_VS_DLY_1                0x45
+#define S_VS_H_2                  0x47
+#define S_VS_H_1                  0x48
+#define S_VS_H_0                  0x49
 #define S_RSVD_97                 0x97
 
 struct max9286_register_write {
diff --git a/drivers/media/i2c/max9286.c b/drivers/media/i2c/max9286.c
index bf05072..06bbba3 100644
--- a/drivers/media/i2c/max9286.c
+++ b/drivers/media/i2c/max9286.c
@@ -233,6 +233,31 @@ static int max9286_set_stream(struct v4l2_subdev *subdev, int enable)
 		S_ADDR_MAX96705, S_CMLLVL_PREEMP, 0xAA);
 	usleep_range(5000, 6000);
 
+	/* Set VSYNC Delay */
+	max96705_write_register(max, S_ADDR_MAX96705_BROADCAST -
+		S_ADDR_MAX96705, S_SYNC_GEN_CONFIG, 0x21);
+	usleep_range(5000, 6000);
+
+	max96705_write_register(max, S_ADDR_MAX96705_BROADCAST -
+		S_ADDR_MAX96705, S_VS_DLY_2, 0x06);
+	usleep_range(5000, 6000);
+
+	max96705_write_register(max, S_ADDR_MAX96705_BROADCAST -
+		S_ADDR_MAX96705, S_VS_DLY_1, 0xD8);
+	usleep_range(5000, 6000);
+
+	max96705_write_register(max, S_ADDR_MAX96705_BROADCAST -
+		S_ADDR_MAX96705, S_VS_H_2, 0x26);
+	usleep_range(5000, 6000);
+
+	max96705_write_register(max, S_ADDR_MAX96705_BROADCAST -
+		S_ADDR_MAX96705, S_VS_H_1, 0x00);
+	usleep_range(5000, 6000);
+
+	max96705_write_register(max, S_ADDR_MAX96705_BROADCAST -
+		S_ADDR_MAX96705, S_VS_H_0, 0x00);
+	usleep_range(5000, 6000);
+
 	/* Enable link equalizers */
 	rval = regmap_write(max->regmap8, DS_ENEQ, 0x0F);
 	if (rval) {
-- 
2.7.4

